pkg/controller/clustersync/clustersync_controller.go

```diff
@@ -626,7 +658,7 @@ func (r *ReconcileClusterSync) applySyncSets(
 			newSyncStatus.FailureMessage = err.Error()
 		}
 		if syncSetNeedsRequeue {
-			requeue = true
+			requeue = requeueWithDelay
```

Comment:
So IIUC you're delaying if apply/create/update fails, but requeueing immediately if a delete fails? Can you explain the rationale?

I think if we're going to be picking and choosing, it should be based on the nature of the error... which can be hard to get right. For example, if we could detect that the syncset itself is bogus, we should arguably mark it as such and not even retry. If the failure is that the target cluster is unreachable, we should retry, probably with a delay, but what should that delay be? What about a creds failure? Perhaps we don't requeue, making the user fix their creds and force resync by deleting the lease (or waiting until the next 2h interval).